pkg/vsphere/tags/categories.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tags
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/juju/errors"
+)
+
+const (
+	CategoryURL = "/com/vmware/cis/tagging/category"
+)
+
+type CategoryCreateSpec struct {
+	CreateSpec CategoryCreate `json:"create_spec"`
+}
+
+type CategoryCreate struct {
+	AssociableTypes []string `json:"associable_types"`
+	Cardinality     string   `json:"cardinality"`
+	Description     string   `json:"description"`
+	Name            string   `json:"name"`
+}
+
+type Category struct {
+	ID              string   `json:"id"`
+	Description     string   `json:"description"`
+	Name            string   `json:"name"`
+	Cardinality     string   `json:"cardinality"`
+	AssociableTypes []string `json:"associable_types"`
+	UsedBy          []string `json:"used_by"`
+}
+
+func (c *RestClient) CreateCategoryIfNotExist(ctx context.Context, name string, description string, categoryType string, multiValue bool) (*string, error) {
```

Comment:
Although, since `trace.Logger` is set to output to stderr by default, if stderr for the vic sessions (docker persona, et al) is set to a multi-writer instance that includes the syslog writer, then output from `trace.Operation` log calls will show up in syslog. See https://github.com/vmware/vic/issues/5542